andela/hc-wits-kla
# (WIP) #159165376 Ensure all deployment tasks are running

#### What does this PR do?
* Ensure that all tasks required for the proper execution of the product run on deployment and not manually.

#### Description of Task to be completed?
* Ensure to migrate all migrations if any.
* Ensure that database triggers run as required.
* Ensure that the task for sending alerts to users executes.

#### How should this be manually tested?
* Clone the branch, ```git clone https://github.com/andela/hc-wits-kla/tree/ft-onboarding-content-158175321```
* In your terminal, ```cd hc-wits-kla```
* In the project directory, activate the environment, ```source hc-venv/bin/activate```
* Install requirements ```pip install requirements.txt```
* Make migrations ```python manage.py makemigrations```
* Serve the application ```python manage.py runserver```
* Open another tab in your terminal console either ```bash ./deployment_tasks_script.sh``` or ```sh ./deployment_tasks_script.sh```
* You should be able to see the status of all the tasks that are in the script (i.e. about migrations, triggers, and sending alerts)
* In the browser, visit, 127.0.0.1:8000/ and you should be able to receive alerts without running each of the tasks manually.

#### What are the relevant pivotal tracker stories?
[#159165376](https://www.pivotaltracker.com/story/show/159165376)

#### Screenshots (if appropriate)

Status of the tasks after running the script in the terminal

<img width="1324" alt="screen shot 2018-07-19 at 14 45 56" src="https://user-images.githubusercontent.com/39955231/42941842-aae370c0-8b66-11e8-81dc-535d05d94cf9.png">

<img width="1328" alt="screen shot 2018-07-19 at 14 49 52" src="https://user-images.githubusercontent.com/39955231/42941844-ab10531a-8b66-11e8-83d0-34702368e6b9.png">
